Main menu

BorDekker opened the pull request.

BorDekker committed:
Pause Menu & Main menu

Created a  C# script so that you can pause and resume the game.
Created a test C# script to test the pause function.
Created a main menu scene level so that when you quit the game, you go back to the main menu.

BorDekker committed:
Created 2 C# scripts

Now when your click the start button the camera will move to the gameplay position.
You can pause & resume the game.
You can also go back to the menu and start again.

WaterWalker023 commented on Assets/StartQuitPauseMenu/MainMenu.cs:
```diff
@@ -0,0 +1,42 @@
+    
+    private bool hasClicked = false;
+
+    public TMP_Text titleText;
```
u know what to do

WaterWalker023 commented on Assets/StartQuitPauseMenu/PauseMenu.cs:
```diff
@@ -0,0 +1,47 @@
+
+    public GameObject backToMenuButton;
+    
+    public TMP_Text titleText;
```
same here

WaterWalker023 commented on Assets/StartQuitPauseMenu/TestMove.cs:
```diff
@@ -0,0 +1,13 @@
+
+public class TestMove : MonoBehaviour
+{
+    public Transform target;
```
and here

BorDekker committed:
Correcting pull request review

Made everything that was public an [SerializeField] private in the C# scripts MainMenu, PauseMenu & TestMove

BorDekker committed:
Removed a debug.log

WaterWalker023 commented on Assets/StartQuitPauseMenu/TestMove.cs:
```diff
@@ -0,0 +1,13 @@
+using UnityEngine;
+
+public class TestMove : MonoBehaviour
```
dit script moet weg of een andere naam

BorDekker committed:
Deleted a script that was no longer neccesary

BorDekker committed:
Merge branch 'main' into MainMenu

BorDekker committed:
Merge branch 'main' into MainMenu

BorDekker committed:
Merge branch 'main' into MainMenu

BorDekker committed:
Added my UI work in the main scene

Added UI art from Camilla into my work.
Made camera position so that it comes from above toward below.
With help from Tom I made it so that the start, resume & main menu button work with unity events.

BorDekker committed:
Removed build

WaterWalker023 commented on Assets/StartQuitPauseMenu/MainMenu.cs:
```diff
@@ -0,0 +1,44 @@
+    [SerializeField] private bool hasClicked = false;
+
+    public UnityEvent hasclickt = new();
+    [SerializeField] private UnityEvent _camerathere = new();
```
```suggestion
    public UnityEvent _camerathere = new();
```

WaterWalker023 commented on Assets/StartQuitPauseMenu/MainMenu.cs:
```diff
@@ -0,0 +1,44 @@
+    
+    private void Update()
+    {
+        if (hasClicked)
```
please invert this if statment

WaterWalker023 commented on Assets/StartQuitPauseMenu/PauseMenu.cs:
```diff
@@ -0,0 +1,52 @@
+    // Update is called once per frame
+    void Update()
+    {
+        if (Input.GetKey(KeyCode.Escape))
```
please invert this if statment

WaterWalker023 commented on Assets/player/Movement.cs:
```diff
@@ -1,5 +1,6 @@
 using UnityEditor;
 using UnityEngine;
+using UnityEngine.Events;
```
deze kan weer weg

BorDekker committed:
Did Tom's request to fix for pull request

BorDekker committed:
Merge branch 'main' into MainMenu

BorDekker committed:
Merge branch 'main' into MainMenu

BorDekker committed:
Added code to Pause menu C# script

You can now use both Escape key & resume button to recontinue the game

BorDekker committed:
Made mainMenuPauseMenu a prefab

BorDekker committed:
Merge branch 'main' into MainMenu

BorDekker committed:
Fixed main menu & pause menu
